packages/editor/src/browser/editor-widget.ts

```diff
@@ -29,7 +29,12 @@ export class EditorWidget extends BaseWidget implements SaveableSource, Navigata
         this.addClass('theia-editor');
         this.toDispose.push(this.editor);
         this.toDispose.push(this.editor.onSelectionChanged(() => {
-            if (this.editor.isFocused()) {
+            if (this.editor.isFocused() && this.selectionService.selection !== this.editor) {
```

Comment:
Since we're now using the same code for both the `onSelectionChanged` and `onFocusChanged` listeners, it would be a little DRY-er to break it out into a function named something like `setSelection`.